pybio/spec/utils/build_spec.py

```diff
@@ -0,0 +1,179 @@
+import os
+import datetime
+import hashlib
+import numpy as np
+
+import pybio.spec.raw_nodes as raw_nodes
+
+#
+# utility functions to build the spec from python
+#
+
+
+# TODO refactor common stuff here
+def _build_spec():
+    pass
+
+
+# TODO this should handle both local file paths and urls and download the url somwhere temp
+def _ensure_uri(uri):
+    assert os.path.exists(uri)
+    return uri
+
+
+def _get_hash(path):
+    with open(path, 'rb') as f:
+        data = f.read()
+        return hashlib.sha256(data).hexdigest()
+
+
+# TODO optional
+# TODO type-annotations
+def build_spec_torch(
+    # model specific required
+    source,
+    model_kwargs,
+    weight_uri,
+    test_inputs,
+    test_outputs,
+    # general required
+    name,
+    description,
+    authors,
+    tags,
+    license,
+    documentation,
+    covers,
+    dependecies,
+    # model specific optional
+    sample_inputs=None,
+    sample_outputs=None,
+    # TODO optional arguments to over-ride the input / output tensor descriptions
+    # general optional
+    cite=None,
+    git_repo=None,
+    attachments=None,
+    packaged_by=None,
+    parent=None,
+    run_mode=None,
+    config=None
+):
+    """
+    """
+    model = raw_nodes.Model
```

Comment:
I think model should be a raw_nodes.Model instance...  (same for all other raw_nodes)

Reply:
makes sense, I will fix it.